### ISSUE TITLE
cygwin: path of `include`d files incorrect

### ISSUE THREAD

**Issue by PetrKryslUCSD** [tpapp/julia-repl#74]
Cygwin Emacs 26: The following error is found when `include`ing a buffer.
```
include("/home/PK/thread_buffers.jl");
ERROR: could not open file \home\PK\thread_buffers.jl
Stacktrace:
 [1] include at .\boot.jl:328 [inlined]
 [2] include_relative(::Module, ::String) at .\loading.jl:1105
 [3] include(::Module, ::String) at .\Base.jl:31
 [4] include(::String) at .\client.jl:432
 [5] top-level scope at REPL[23]:1
```
Manually typing in `include("/cygwin64/home/PK/thread_buffers.jl")` succeeds.

**Comment by PetrKryslUCSD** [comment]
Same thing for C-c C-p (change to the folder of the current buffer).

**Comment by PetrKryslUCSD** [comment]
Here is another example, this time accessing a folder on the Windows file system:
```
julia> include("/cygwin64/c/Users/PK/Documents/work/advanced-finite-element-analysis-of-solids/thread_buffers.jl")
ERROR: could not open file \cygwin64\c\Users\PK\Documents\work\advanced-finite-element-analysis-of-solids\thread_buffers.jl
Stacktrace:
 [1] include at .\boot.jl:328 [inlined]
 [2] include_relative(::Module, ::String) at .\loading.jl:1105
 [3] include(::Module, ::String) at .\Base.jl:31
 [4] include(::String) at .\client.jl:432
 [5] top-level scope at REPL[14]:1

shell> ls /cygdrive/c/Users/PK/Documents/work/advanced-finite-element-analysis-of-solids
README.md  thread_buffers.jl  thread_buffers.jl~
```
The file name passed  to `include` is different from the name that the terminal recognizes.

**Comment by PetrKryslUCSD** [comment]
It seems that absolute paths are a problem for Julia running under cygwin. Would be difficult to change the mechanism in julia-repl to run the includes with relative pathnames?

**Comment by tpapp** [comment]
cf https://github.com/JuliaLang/julia/issues/24128

A workaround I can imagine on the julia-repl side is a simple rewriting rule that replaces `"/cygdrive/c/"` with `C:/` or something like that. This should be quite easy to implement, but I need some help from you about the details.

Specifically, it would be nice to have a few examples of

1. `M-x (buffer-file-name)`
2. and the corresponding path that **works** in `include` in Julia

**Comment by PetrKryslUCSD** [comment]
Here are the results of an experiment I did:

First a file located in the cygwin folder structure:
This fails
julia> include("/home/PK/thread_buffers.jl");

This works
julia> include("c:/cygwin64/home/PK/thread_buffers.jl");

Now for a file  located in the Windows folder  structure:
This fails
julia> include("/cygdrive/c/Users/PK/Documents/work/advanced-finite-element-analysis-of-solids/another.jl");

This works
julia> include("c:/Users/PK/Documents/work/advanced-finite-element-analysis-of-solids/another.jl");

Thank you for your help. Much appreciated. I used to use Emacs, and I would like to return to it. (Even though Sublime Text is a pleasure to work with. It is simply not as easy to extend or customize as Emacs.)

**Comment by tpapp** [comment]
Thanks, so in these examples the "this fails" path comes from `(buffer-file-name)`, and the "this works" is what you rewrite it to make it work?

**Comment by PetrKryslUCSD** [comment]
> Thanks, so in these examples the "this fails" path comes from (buffer-file-name), and the "this works" is what you rewrite it to make it work?

That is correct.

**Comment by tpapp** [comment]
(reopening, this will be closed by #76 which is not yet merged)
